ui/app/styles/core/select.scss

```diff
@@ -2,7 +2,61 @@
  * Copyright (c) HashiCorp, Inc.
  * SPDX-License-Identifier: MPL-2.0
  */
+select {
+  appearance: none;
+}
 
 .select select {
+  background-color: $ui-gray-050;
+  box-shadow: 0 3px 1px rgba($black, 0.12);
+  border-color: $grey-light;
+  border-radius: 2px;
+  color: $grey-dark;
+  height: 2.5rem;
+  padding-bottom: calc(0.375em - 1px);
+  padding-left: $size-8;
+  padding-right: $size-8;
+  padding-top: calc(0.375em - 1px);
   text-rendering: auto !important;
+
+  @include until($desktop) {
+    font-size: 16px;
+  }
+  &::placeholder {
+    opacity: 0.5;
+  }
+
+  .has-background-grey-lighter & {
+    background-color: $white;
+  }
+}
+
+.select select[disabled] {
+  border-color: $grey-light;
+  background-color: $ui-gray-100;
+  color: $ui-gray-500;
+  &:hover {
+    border-color: $grey-light;
+  }
+}
+
+.select {
+  height: 2.5rem;
+}
+
+.select::after {
```

Comment:
handles the icon on the right side. 
![image](https://user-images.githubusercontent.com/6618863/227725008-302462c7-312c-4d04-80a7-36e0c9111228.png)